src/vs/workbench/contrib/bulkEdit/browser/bulkTextEdits.ts

```diff
@@ -156,7 +156,7 @@ class EditorEditTask extends ModelEditTask {
 						});
 					}
 				}
-				snippetCtrl.apply(snippetEdits, { undoStopBefore: false, undoStopAfter: false });
+				snippetCtrl.apply(snippetEdits, { undoStopBefore: false, undoStopAfter: false, adjustWhitespace: false });
```

Comment:
This seems like an unrelated and unwanted change

Reply:
Just to explain why I added this. Yes, the main change is about adding an optional flag to `insertSnippet`, but I noticed that when a provided code action is applied, it doesn't go through the text editor `insertSnippet`. It seems to directly call the `SnippetController` here. This is what affects the actual behaviour for all extensions providing code actions. So any extension with a code action provider that includes snippets (eg. https://github.com/redhat-developer/vscode-java/issues/3686), and don't want VS Code re-adjusting indentation won't be resolved without this change. I also thought I could change the default behaviour (adjust indentation for snippets) based on comments like https://github.com/microsoft/language-server-protocol/pull/1892#issuecomment-1964186341 .

If this change to the default isn't wanted, would you support (in this PR) adding a field to `vscode.CodeAction`, `keepWhitespace`, much like is done for `vscode.CompletionItem`, and have it be honoured at https://github.com/microsoft/vscode/blob/12b060f0a812fcd7e4cabf9c65e8267f48519459/src/vs/editor/contrib/codeAction/browser/codeAction.ts#L299-L306 .

Reply:
I understand but also here we cannot change the existing behaviour. I think this would be `keepWhitespace` for the `SnippetTextEdit` type